pyzim/pyzim.pyx

```diff
@@ -1,338 +1,183 @@
-        """Get the article's content"""
-        return self.c_zim_article.content.decode('UTF-8')
+    def get_url(self):
+        raise NotImplementedError
 
-    @content.setter
-    def content(self, new_content):
-        """Set the article's content"""
-        self.c_zim_article.content = new_content.encode('UTF-8') 
+    def get_title(self):
+        raise NotImplementedError
+    
+    def is_redirect(self):
+        raise NotImplementedError
 
-    @property
-    def longurl(self):
-        """Get the article's long url i.e {NAMESPACE}/{url}"""
-        return self.c_zim_article.getUrl().getLongUrl().decode("UTF-8", "strict")
+    def get_mime_type(self):
+        raise NotImplementedError
 
-    @property
-    def url(self):
-        """Get the article's url"""
-        return self.c_zim_article.url.decode('UTF-8')
-    
-    @url.setter
-    def url(self, new_url):
-        """Set the article's url"""
-        self.c_zim_article.url = new_url.encode('UTF-8')
+    def get_filename(self):
+        raise NotImplementedError
 
-    @property
-    def redirect_longurl(self):
-        """Get the article's redirect long url i.e {NAMESPACE}/{redirect_url}"""
-        return self.c_zim_article.getRedirectUrl().getLongUrl().decode("UTF-8", "strict")
+    def should_compress(self):
+        raise NotImplementedError
+
+    def should_index(self):
+        raise NotImplementedError
 
-    @property
     def redirect_url(self):
-        """Get the article's redirect url"""
-        return self.c_zim_article.redirectUrl.decode('UTF-8')
+        raise NotImplementedError
 
-    @redirect_url.setter
-    def redirect_url(self, new_redirect_url):
-        """Set the article's redirect url"""
-        self.c_zim_article.redirectUrl = new_redirect_url.encode('UTF-8')
+    def get_data(self):
+        raise NotImplementedError
 
     @property
     def mimetype(self):
-        """Get the article's mimetype"""
-        return self.c_zim_article.mimeType.decode('UTF-8')
-
-    @mimetype.setter
-    def mimetype(self, new_mimetype):
-        """Set the article's mimetype"""
-        self.c_zim_article.mimeType = new_mimetype.encode('UTF-8')
-
+        return self.c_article.getMimeType().decode('UTF-8')
+    
     @property
-    def is_redirect(self):
-        """Get if the article is a redirect"""
-        return self.c_zim_article.isRedirect()
+    def content(self):
+        blob = self.c_article.getData()
+        return blob.data()[:blob.size()].decode('UTF-8')
 
+    # This changes with implementation
     @property
     def can_write(self):
-        """Get if the article is valid for writing"""
-        # An article must have at least non empty url to be writable
-        # Content can be empty if article is a redirect
-        if self.longurl and (self.content or self.is_redirect):
-            self._can_write = True
-        else:
-            self._can_write = False
-        return self._can_write
+        raise NotImplementedError
```

Comment:
This should not be necessary has this is the type of the Article who define if we can use it to write a zim file.

(It IS not necessary, but maybe for technical reason as handling python ducktyping you may want a marker. But it should not be to the (user) implementation to change this).

Reply:
error.